src/lib.rs

```diff
@@ -622,13 +630,54 @@ impl GenerateTasksService for GenerateTasksServiceImpl {
+                                BURN_IN_TAG_BUILD_VARIANTS,
+                                build_variant,
+                            );
+                        let compile_distro = self
+                            .evg_config_utils
+                            .lookup_build_variant_expansion(
+                                BURN_IN_TAG_COMPILE_DISTRO,
+                                build_variant,
+                            ).unwrap_or_else(|| {
+                                panic!(
+                                    "`{}` is missing the burn_in_tag compile distro expansion. Set the '{}' expansion in your project's config to continue.",
+                                    build_variant.name, BURN_IN_TAG_COMPILE_DISTRO
+                                )
+                            });
+                        let compile_task_group_name = self
+                            .evg_config_utils
+                            .lookup_build_variant_expansion(
+                                BURN_IN_TAG_COMPILE_TASK_GROUP_NAME,
+                                build_variant,
+                            ).unwrap_or_else(|| {
+                                panic!(
+                                    "`{}` is missing the burn_in_tag compile task group name expansion. Set the '{}' expansion in your project's config to continue.",
+                                    build_variant.name, BURN_IN_TAG_COMPILE_TASK_GROUP_NAME
+                                )
+                            });
+                        for variant in burn_in_tag_build_variants {
+                            let bv_info = burn_in_tag_build_variant_info
+                                .entry(variant.clone())
+                                .or_insert(BurnInTagBuildVariantInfo {
+                                    compile_distro: compile_distro.clone(),
+                                    compile_task_group_name: compile_task_group_name.clone(),
+                                });
+                            if bv_info.compile_distro != compile_distro.clone() {
+                                panic!(
+                                    "`{}` is trying to set a different compile distro than already exists for {}. Check the {} expansions in your config.",
+                                    build_variant.name, variant, BURN_IN_TAG_COMPILE_DISTRO
+                                )
+                            }
+                            if bv_info.compile_task_group_name != compile_task_group_name {
+                                panic!(
+                                    "`{}` is trying to set a different compile task group name than already exists for {}. Check the {} expansions in your config.",
+                                    build_variant.name, variant, BURN_IN_TAG_COMPILE_TASK_GROUP_NAME
+                                )
+                            }
+                        }
```

Comment:
Is it possible to extract this part to a separate function? `generate_build_variants` is getting very huge, I wonder if we can make it somewhat better.